astropy/utils/compat/numpycompat.py

```diff
@@ -21,3 +22,4 @@
 NUMPY_LT_1_22_1 = not minversion(np, '1.22.1')
 NUMPY_LT_1_23 = not minversion(np, '1.23')
 NUMPY_LT_1_24 = not minversion(np, '1.24dev0')
+NUMPY_GE_1_24 = minversion(np, '1.24dev0')
```

Comment:
Why not reuse `NUMPY_LT_1_24` instead of defining a new var?

Reply:
not NUMPY_LT_1_24 didn't work for me with the `skipif` decorator

Reply:
That makes no sense. Only difference between `NUMPY_LT_1_24` and `NUMPY_GE_1_24` is the `not`. 🤯

Reply:
Do you think we need to do something crazy like this to make it work?

```python
NUMPY_LT_1_24 = not minversion(np, '1.23.999')
```

Reply:
Have you seen this error?
```
________________ ERROR at setup of TestSortFunctions.test_msort ________________
Error evaluating 'skipif': you need to specify reason=STRING when using booleans as conditions.
```

Reply:
Yes, that has nothing to do with the var. You provide an optional reason.

```python
@pytest.mark.skipif(not NUMPY_LT_1_24, reason='blah blah')
```

Reply:
I updated the PR.  Tests now pass locally for me.